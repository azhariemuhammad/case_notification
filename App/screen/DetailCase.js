import React, {useState, useEffect} from 'react'
import {Container, Content, Text} from "native-base";

import {baseService} from "../services";
import {StyleSheet, View} from "react-native";

const DetailCase = (props) => {

    const [caseInfo, setCaseInfo] = useState({});

    useEffect(() => {
        const {navigation} = props;
        const caseId = navigation.getParam('caseId');

        async function getCaseInfo() {
            await baseService().getCaseInformation(caseId).then(result => {
                setCaseInfo(result.data)
            }).catch(e => {
                setCaseInfo({})
            })

        }

        getCaseInfo();
    }, []);

    const {
        name, address,
        gender, isPregnant,
        age, diseaseType, 
        classificationCase,
        patientContact,
        reporter
    } = caseInfo
    return (
        <Container>
            <Content>
                <View style={styles.inputHeight}>
                    <Text style={styles.textSecondary}>Nama</Text>
                    <Text style={styles.textBody}>{name}</Text>
                </View>
                <View style={styles.inputHeight}>
                    <Text style={styles.textSecondary}>Alamat</Text>
                    <Text style={styles.textBody}>{address}</Text>
                </View>
                <View style={styles.inputHeight}>
                    <Text style={styles.textSecondary}>Jenis Kelamin</Text>
                    <Text style={styles.textBody}>{gender}</Text>
                </View>
                {
                    (gender === 'Women' && age > 17 ) &&
                    <View style={styles.inputHeight}>
                        <Text style={styles.textSecondary}>Sedang Hamil</Text>
                        <Text style={styles.textBody}>{(isPregnant) ? 'Ya' : 'Tidak'}</Text>
                    </View>
                }
                <View style={styles.inputHeight}>
                    <Text style={styles.textSecondary}>Umur</Text>
                    <Text style={styles.textBody}>{age}</Text>
                </View>
                <View style={styles.inputHeight}>
                    <Text style={styles.textSecondary}>Tipe Malaria</Text>
                    <Text style={styles.textBody}>{diseaseType}</Text>
                </View>
                <View style={styles.inputHeight}>
                    <Text style={styles.textSecondary}>Kontak Pasien</Text>
                    <Text style={styles.textBody}>{patientContact}</Text>
                </View>
                <View style={{height: 70, marginBottom: 20}}>
                    <Text style={styles.textSecondary}>Pelapor</Text>
                    <Text style={styles.textBody}>{reporter}</Text>
                </View>
            </Content>
        </Container>

    )
}


const styles = StyleSheet.create({
    textSecondary: {
        color: '#000000',
        fontSize: 18
    },
    inputHeight: {
        height: 70,
    },
    textBody: {
        padding: 8
    },
    wrapperDetail: {
        flex: 1
    }
})

export default DetailCase
