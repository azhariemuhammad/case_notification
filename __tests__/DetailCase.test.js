import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';

import DetailCase from "../App/screen/DetailCase";


it("should render DetailCase component and compare with snapshot", () => {
    const wrapper = shallow(<DetailCase/>)
    expect(wrapper).toMatchSnapshot()
});
