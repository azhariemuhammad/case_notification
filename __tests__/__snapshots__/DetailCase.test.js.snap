// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DetailCase component and compare with snapshot 1`] = `ShallowWrapper {}`;
